### PR TITLE
Several fixes when setting revision

### PR DESCRIPTION
**Pull request by massonju**
- return right revision when git ls-remote
- set revision according to remote url and branch

### CODE REVIEW

**Comment by massonju** [comment]
Cc @david-baylibre

File: repo_resource/common.py
Comment (makohoek) [inline repo_resource/common.py:99]: Please find a better variable name.
`branchOrig` honestly does not means much.

I think I understand the problem but I don't think the current code is clear enough to explicit this issue.

File: repo_resource/common.py
Comment (makohoek) [inline repo_resource/common.py:104]: Does the comment needs to be updated?

File: repo_resource/common.py
Comment (makohoek) [inline repo_resource/common.py:478]: This part definitely deserves a comment. It might seem obvious but we will forget about this.

File: repo_resource/common.py
Comment (makohoek) [inline repo_resource/common.py]: nitpick: commit message: Remove "this patch" and just use imperative mood.

File: repo_resource/common.py
Comment (makohoek) [inline repo_resource/common.py:117]: This code seems to be moved from the other commit.
Why is this now needed ?

**Review by makohoek — CHANGES_REQUESTED** [review]

File: repo_resource/common.py
Comment (massonju) [reply repo_resource/common.py:104]: yes correct I will update this comment

File: repo_resource/common.py
Comment (massonju) [reply repo_resource/common.py:478]: ok

File: repo_resource/common.py
Comment (massonju) [reply repo_resource/common.py:117]: Well no this code is needed to fix the issue described in this commit.

It's not related to the other commits

File: repo_resource/common.py
Comment (massonju) [reply repo_resource/common.py]: yes

File: repo_resource/common.py
Comment (massonju) [reply repo_resource/common.py:117]: By converting the revision list to revision dict with `refs/heads/XXXX` as key, we ensure that the right revision is returned

File: repo_resource/common.py
Comment (massonju) [reply repo_resource/common.py]: I've changes the logic in this commit:
https://github.com/makohoek/repo-resource/pull/40/commits/ec10c92a0764ffe685dbe03e21120c976a9958ec

`branchOrig` is not present anymore

**Review by makohoek — APPROVED** [review]

File: repo_resource/common.py
Comment (makohoek) [reply repo_resource/common.py]: Looks better, thank you.

**Comment by makohoek** [comment]
I'd like to have another pair of eyes on this review, so we will be awaiting for @david-baylibre 's feedback on this before merging.

File: repo_resource/common.py
Comment (makohoek) [reply repo_resource/common.py:99]: I've turned this into an issue instead, to continue the discussion.

Since this has been approved, I'll go ahead and merge and draft a new minor release.

@massonju please continue discussion the review comments here or in the dedicated issue: https://github.com/makohoek/repo-resource/issues/42

**Pull request merged**
